lib/telephone_number/parser.rb

```diff
@@ -7,33 +7,65 @@ def sanitize(input_number)
       return input_number.gsub(/[^0-9]/, "")
     end
 
-    def extract_number_types(input_number, country)
-      country_data = TelephoneNumber::PhoneData.phone_data[country.to_sym]
+    # returns an array of valid types for the normalized number
+    # if array is empty, we can assume that the number is invalid
+    def validate
+      return [] unless country_data
+      applicable_keys = country_data[TelephoneNumber::PhoneData::VALIDATIONS].reject{ |key, _value| KEYS_TO_SKIP.include?(key) }
```

Comment:
I don't think so. We don't need the values in this statement, but we do use the values when we go to iterate.